crates/sui-bridge/src/orchestrator.rs

```diff
@@ -140,12 +138,16 @@ where
         executor_tx: mysten_metrics::metered_channel::Sender<BridgeActionExecutionWrapper>,
         mut eth_events_rx: mysten_metrics::metered_channel::Receiver<(
             ethers::types::Address,
+            u64,
             Vec<EthLog>,
         )>,
     ) {
         info!("Starting eth watcher task");
-        while let Some((contract, logs)) = eth_events_rx.recv().await {
+        while let Some((contract, end_block, logs)) = eth_events_rx.recv().await {
             if logs.is_empty() {
+                store
+                    .update_eth_event_cursor(contract, end_block)
+                    .expect("Store operation should not fail");
```

Comment:
meat 2 of this PR - Eth syncer's cursor is updated even when the event list is empty

Reply:
If the `if logs.is_empty() { ... }` block is removed, `update_eth_event_cursor()` will still be called later, right?

Reply:
correct! I put the block here in case we add something in the future that accidentally breaks the logic